File: advent-of-code-2023/src/test/kotlin/aoc/snowrescuemission/Day04Test.kt
package aoc.snowrescuemission

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test

class Day04Test {

    @Test
    fun testPart1() {
        Day04("input/day04.txt").part1() shouldBe 24_848
    }

    @Test
    fun testPart2() {
        Day04("input/day04.txt").part2() shouldBe 7_258_152
    }
}

File: advent-of-code-2024/dart/test/day04_test.dart
import 'package:test/test.dart';

import '../bin/day04.dart';

void main() {

  var day04 = Day04.using("../input/day04.txt");

  test('test day 4 part 1', () {
    expect(day04.part1(), 2662);
  });

  test('test day 4 part 2', () {
    expect(day04.part2(), 2034);
  });
}
